src/Bicep.Core/Modules/OciModuleReference.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Configuration;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Registry.Oci;
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+
+namespace Bicep.Core.Modules
+{
+    /// <summary>
+    /// Represents a reference to an artifact in an OCI registry.
+    /// </summary>
+    public class OciModuleReference : ModuleReference, IOciArtifactReference
+    {
+        public string Registry => this.ociArtifactRef.Registry;
+        public string Repository => this.ociArtifactRef.Repository;
+        public string? Tag => this.ociArtifactRef.Tag;
+        public string? Digest => this.ociArtifactRef.Digest;
+        public string ArtifactId => this.ociArtifactRef.ArtifactId;
```

Comment:
nit: It'd better to put properties after the constructor and overridden properties.